pkg/loader/http.go

```diff
@@ -80,6 +91,10 @@ func NewHTTPURLLoader(skipTLS bool, cache cache.Cache) (*HTTPURLLoader, error) {
 	retryClient.HTTPClient = &http.Client{Transport: transport}
 	retryClient.Logger = nil
 
-	httpLoader := HTTPURLLoader{client: *retryClient.StandardClient(), cache: cache}
+	httpLoader := HTTPURLLoader{
+		client:    *retryClient.StandardClient(),
+		cache:     cache,
+		authToken: authToken,
```

Comment:
definitely need to be sure we only pass the token to Github, and not leak it to other registries